docs/DeveloperGuide.md

```diff
@@ -217,54 +230,87 @@ The `ViewCommandParser` uses ``ViewCommandParser#arePrefixesPresent`` to check f
 
 ### View Stats feature
 This feature supports the viewing of summary statistics, it currently shows the total number of students, the total amount
 owed by them and the number of exams in the upcoming months.
 
 #### Implementation
-It's mechanism is similar to the `help` feature where a popout window is shown when called.
+Its mechanism is similar to the `help` feature where a popout window is shown when called.
 
-It gets it's info from the `logic` interface
+It gets its info from the `logic` interface
 
-### View Name Feature
-** to add in v1.4**
-#### Implementation
-** to add in v1.4**
-### View ID Feature
-** to add in v1.4**
+### Add Log feature
+This command is has a similar mechanism to the `add` feature, but targets a specific student instead. <br>
+This feature enables tutors to log session specific details for record to a specific student. Each log entry includes the total hours of the lesson, lesson content, the learning style of the student, as well as any additional notes. The date of the log entry is recorded as the system time when the user added the log.
 #### Implementation
-** to add in v1.4**
-
-# Student Payment Management System
+The parser first checks if the there exists a student with the `ID` specified using the `-id` in current records. Then, the app adds the log entry to the end of the log list attached as a field to the student.
+Below is the sequence diagram of how the `log` command interacts with multiple classes.
+<puml src="diagrams/LogSequenceDiagram.puml">
+#### Design Considerations
+**Aspect: Whether all fields in log should be compulsory**
+* **Alternative 1 (current choice):** All fields are compulsory
+    * Pros: Reminds user of all possible fields, gives structure to the display format of the log entries.
+    * Cons: Troublesome for user to enter all fields, mandates some content even if they deem it unnecessary
+* **Alternative 2:** Only some fields like hours are compulsory
+    * Pros: Allows flexibility in user input, saves time for user
+    * Cons: Log entries will not be structured. Tutor may also accidentally forget important information to log, such as a change in learning style or lesson content covered.
+      As we are currently in the early stage of development and have yet to carry out user testing for what fields should be compulsory, we chose to mandate every field so tutors are reminded of all possibly important aspects to record.
+      **Aspect: Restrictions on log feature fields' contents**
+* **Alternative 1 (current choice):** All fields are strings, and can be empty
+    * Pros: Easy to implement, flexible, allowing customised inputs like `45 minutes` for hours instead of restricting to an integer. Allowing empty fields partially mitigates aforementioned con of being forced to enter content even if the user deems it unnecessary, all while still providing a reminder to possibly important fields.
+    * Cons: Prone to user errors, which can make the log entry messy.
+
```

Comment:
Hmm, i think Alternative 2 is missing for this aspect.

Reply:
OH OOps will fix that shortly!